src/map.ts

```diff
@@ -73,7 +73,7 @@ export async function mapLimit<V>(input: any, limit: number, iteratee: any): Pro
     return [];
   }
 
-  const isArray = input.length !== undefined;
+  const isArray = input instanceof Array;
```

Comment:
`Array.isArray()` exists

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray

Reply:
http://web.mit.edu/jwalden/www/isArray.html - this article was particularly interesting in regards to that. I tend to agree w/ @dhermes that if we're going to fix it, let's fix it whole hog.

Reply:
I agree with you both, and we should fix the whole hog.